packages/sarif-to-comment/src/cli.ts

```diff
@@ -147,6 +164,19 @@ export function run() {
         if (emptyURLReasons.length > 0) {
             console.log("Some comments were not posted, reasons will be included");
         }
+        const shouldFailResults = commentsResults.reduce((acc: boolean, result: any) => {
+            return result.shouldFail || acc; // always returns true if there is one of the map values is true
+        }, false);
+
+        if (shouldFailResults) {
+            console.log(
+                postedURLS.concat(emptyURLReasons).join("\n") +
+                    "\nFailing ! An issue with severity " +
+                    cli.flags.failon +
+                    " was found."
+            );
+            process.exit(1);
+        }
```

Comment:
https://github.com/security-alert/security-alert/blob/master/packages/sarif-to-comment/bin/cmd.js
cli entry catch promise rejection and exit with status 1.
So, we can throw error instead of exit on here.

```js
const errorMessage = postedURLS.concat(emptyURLReasons).join("\n") +
                    "\nFailing ! An issue with severity " +
                    cli.flags.failon +
                    " was found."
throw new Error(errorMessage);
```

(I'm not so confident about this code.)

📝 [Template literals](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals) is useful for readble message.

Reply:
Done